### PR TITLE
Update Improving Performance Guide

### LINKED ISSUE #1571: Update Improving Computational Performance Page with Woodwork Dataframe
- The page below should be updated with the new function calls, parameters, and the accessor approach.
- https://featuretools.alteryx.com/en/latest/guides/performance.html
- We should also at the same time **use Jupyter notebooks, rather than RST files.**

### PR DESCRIPTION
**Pull request by frances-h**
- Convert performance.rst to a jupyter notebook
- Update code samples to include woodwork specific changes

Closes #1571

### CODE REVIEW

File: docs/source/guides/performance.ipynb [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+   "cell_type": "markdown",
+   "id": "9dab133a",
+   "metadata": {},
+   "source": [
+    "# Improving Computational Performance\n",
+    "\n",
+    "Feature engineering is a computationally expensive task. While Featuretools comes with reasonable default settings for feature calculation, there are a number of built-in approaches to improve computational performance based on dataset and problem specific considerations.\n",
+    "\n",
+    "## Reduce number of unique cutoff times\n",
+    "Each row in a feature matrix created by Featuretools is calculated at a specific cutoff time that represents the last point in time that data from any entity in an entity set can be used to calculate the feature. As a result, calculations incur an overhead in finding the subset of allowed data for each distinct time in the calculation."
```
Comment (tamargrey) [inline docs/source/guides/performance.ipynb:13]: entity usage! Should replace with DataFrame

File: docs/source/guides/performance.ipynb [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+   "cell_type": "raw",
+   "id": "abd2207f",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    ".. note::\n",
+    "    Support for Dask EntitySets and Koalas EntitySets is still in Beta. While the key functionality has been implemented, development is ongoing to add the remaining functionality.\n",
+    "\n",
+    "    All planned improvements to the Featuretools/Dask and Featuretools/Koalas integration are documented on Github (`Dask issues <https://github.com/alteryx/featuretools/issues?q=is%3Aopen+is%3Aissue+label%3ADask>`_, `Koalas issues <https://github.com/alteryx/featuretools/issues?q=is%3Aopen+is%3Aissue+label%3AKoalas>`_). If you see an open issue that is important for your application, please let us know by upvoting or commenting on the issue. If you encounter any errors using Dask or Koalas entities, or find missing functionality that does not yet have an open issue, please create a `new issue on Github <https://github.com/alteryx/featuretools/issues>`_."
```
Comment (tamargrey) [inline docs/source/guides/performance.ipynb:52]: entities > "DataFrames"

File: docs/source/guides/performance.ipynb [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  },
+  {
+   "cell_type": "raw",
+   "id": "94c74ef7",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    ".. note::\n",
+    "    Partitioning data is not necessary when using a Dask ``EntitySet``, as the Dask dataframes that make up the ``EntitySet`` are already partitioned. Partitioning is only needed when working with pandas entities."
```
Comment (tamargrey) [inline docs/source/guides/performance.ipynb:168]: entities -> DataFrames

Comment (thehomebrewnerd) [reply docs/source/guides/performance.ipynb:168]: We might also want to update this note to mention Koalas along with Dask.

**Review by thehomebrewnerd — COMMENTED** [review]
The only other thing we might want to optionally clean up here is that there are a number of places where we use `entity set` and some where we use `entityset`. Even though it isn't directly related to this PR, it might be good to be consistent and replace all instances of `entity set` with `entityst.

**Review by thehomebrewnerd — APPROVED** [review]
LGTM!

**Pull request merged**